client/src/js/settings/components/Jobs/Resources.js

```diff
@@ -1,59 +1,193 @@
-                    withButton
-                />
-                <InputError
-                    label="Memory Limit (GB)"
-                    type="number"
-                    onSave={props.onUpdateMem}
-                    initialValue={props.mem}
-                    withButton
-                />
-            </Panel>
-        </Col>
-    </Row>
-);
+import { getResources } from "../../../jobs/actions";
+import { InputError, LoadingPlaceholder, Icon} from "../../../base";
+
+class Resources extends React.Component {
+
+    constructor (props) {
+        super(props);
+
+        this.state = {
+            errorProc: false,
+            errorMem: false,
+            procUpperLimit: this.props.proc,
+            memUpperLimit: this.props.mem,
+            showAlert: false
+        };
+    }
+
+    componentDidMount () {
+        this.props.onGet();
+    }
+
+    componentWillReceiveProps (nextProps) {
+        const procLimit = nextProps.resources.proc.length;
+        const memLimit = parseFloat((nextProps.resources.mem.total / Math.pow(1024, 3)).toFixed(1));
+
+        if (nextProps.mem !== this.props.mem) {
+            this.setState({errorMem: false});
+        }
+
+        if (nextProps.proc !== this.props.proc) {
+            this.setState({errorProc: false});
+        }
+
+        if (!this.props.resources && nextProps.resources) {
+            this.setState({
+                procUpperLimit: procLimit,
+                memUpperLimit: memLimit
+            });
+
+            if (procLimit < this.props.proc) {
+                this.props.onUpdateProc({ value: procLimit });
+            }
+
+            if (memLimit < this.props.mem) {
+                this.props.onUpdateMem({ value: memLimit });
+            }
+        }
+    }
+
+    handleChangeProc = (e) => {
+        e.preventDefault();
+        this.setState({ errorProc: false });
+    };
+
+    handleChangeMem = (e) => {
+        e.preventDefault();
+        this.setState({ errorMem: false });
+    };
+
+    handleInvalidProc = (e) => {
+        e.preventDefault();
+        this.setState({ errorProc: true });
+    };
+
+    handleInvalidMem = (e) => {
+        e.preventDefault();
+        this.setState({ errorMem: true });
+    };
+
+    handleSaveProc = (e) => {
+        if (e.value <= this.state.procUpperLimit && e.value >= this.props.procLowerLimit) {
+            this.props.onUpdateProc(e);
+        } else {
+            this.setState({ errorProc: true });
+        }
+    };
+
+    handleSaveMem = (e) => {
+        if (e.value <= this.state.memUpperLimit && e.value >= this.props.memLowerLimit) {
+            this.props.onUpdateMem(e);
+        } else {
+            this.setState({ errorMem: true });
+        }
+    };
+
+    render () {
+        if (this.props.resources === null) {
+            return <LoadingPlaceholder />;
+        }
+
+        const errorMessageProc = this.state.errorProc ? "Cannot go over or under resource limits" : null;
```

Comment:
Can you reformat this message to "Value must be between {min} and {max}"?

This logic can be put into a separate function defined at the top of the module so that it can be reused for both the `proc` and `mem` values.